src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,135 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
-    public void add(T value) {
+    public Object[] toArray() {
+        return Arrays.copyOf(elementData, size);
+    }
 
+    @Override
+    public void add(T value) {
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        System.arraycopy(elementData, index,
+                        elementData, index + 1,
+                        size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] arrayToAdd = list.toArray();
+        int arrayToAddLength = arrayToAdd.length;
+        if (arrayToAddLength > elementData.length - size) {
+            grow(size + arrayToAddLength);
+        }
+        System.arraycopy(arrayToAdd, 0, elementData, size, arrayToAddLength);
+        size += arrayToAddLength;
```

Comment:
let's simplify your code here. I'd recommend iterating through the list and using your add() method